#! /bin/sh /usr/share/dpatch/dpatch-run

## 20_vdr-1.3.31-sequence-end-code5.dpatch by Reinhard Nissl <rnissl@gmx.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes a problem when moving cutting marks in radio recordings

@DPATCH@
diff -urNad vdr-1.3.31/dvbplayer.c /tmp/dpep.hVSHGH/vdr-1.3.31/dvbplayer.c
--- vdr-1.3.31/dvbplayer.c	2005-08-14 12:52:45.000000000 +0200
+++ /tmp/dpep.hVSHGH/vdr-1.3.31/dvbplayer.c	2005-08-31 21:26:31.886530960 +0200
@@ -661,7 +661,7 @@
            if (playMode == pmPause)
               DevicePlay();
            // append sequence end code to get the image shown immediately with softdevices
-           if (r > 6) { // should be always true
+           if (r > 6 && (b[3] & 0xF0) == 0xE0) { // make sure to append it only to a video packet
               b[r++] = 0x00;
               b[r++] = 0x00;
               b[r++] = 0x01;
